include/engine/internal_route_result.hpp

```diff
@@ -145,9 +145,22 @@ inline InternalRouteResult CollapseInternalRouteResult(const InternalRouteResult
             collapsed.target_traversed_in_reverse.back() =
                 leggy_result.target_traversed_in_reverse[i];
             // copy path segments into current leg
-            last_segment.insert(last_segment.end(),
-                                leggy_result.unpacked_path_segments[i].begin(),
-                                leggy_result.unpacked_path_segments[i].end());
+            if (!leggy_result.unpacked_path_segments[i].empty())
+            {
+                auto old_size = last_segment.size();
+                last_segment.insert(last_segment.end(),
+                                    leggy_result.unpacked_path_segments[i].begin(),
+                                    leggy_result.unpacked_path_segments[i].end());
+
+                // The first segment of the unpacked path is missing the weight of the
+                // source phantom.  We need to add those values back so that the total
+                // edge weight is correct
+                last_segment[old_size].weight_until_turn +=
+                    leggy_result.segment_end_coordinates[i].source_phantom.forward_weight;
```

Comment:
if the source phantom traversed in reverse then the reverse weight must be used:
```
leggy_result.source_traversed_in_reverse[i] ?
leggy_result.segment_end_coordinates[i].source_phantom.reverse_weight :
leggy_result.segment_end_coordinates[i].source_phantom.forward_weight;
```